### PR TITLE
Simplify recommendations endpoint

### PR DESCRIPTION
**Pull request by marioortizmanero**
## Description

The recommendations endpoint was very messy and had a long-standing TODO item. I've simplified it so that it just takes a map with the attributes that are directly appended to the query payload.

## Dependencies 

None

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

The `test_recommendations` endpoint

### CODE REVIEW

File: src/clients/base.rs [hunk 1]
```diff
@@ -901,7 +906,7 @@ where
     /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-recommendations)
     async fn recommendations<'a>(
         &self,
-        payload: &Map<String, Value>,
+        attributes: impl IntoIterator<Item = (&'a str, Value)> + Send + 'a,
```
Comment (ramsayleung) [inline src/clients/base.rs:909]: Is it better to make attribute as a named type(such as `Enum`) instead of a string?

Comment (marioortizmanero) [reply src/clients/base.rs]: Done. I wanted to avoid the huge enum but it's indeed much easier to use this way.

**Comment by ramsayleung** [comment]
LGTM, since you have changed the signature of this endpoint, would you like to update the **break change** part of `CHANGELOG.md`

**Comment by marioortizmanero** [comment]
Sure!

**Review by ramsayleung — APPROVED** [review]

**Pull request merged**